packages/vscode/src/plugins/prisma-postgres-manager/management-api/client.ts

```diff
@@ -7,11 +7,69 @@ class FetchError extends Error {
   }
 }
 
-export const createManagementAPIClient = (token: string) => {
-  const client = createClient<paths>({ baseUrl: 'https://api.prisma.io' })
+/**
+ * Creates a fetch function that includes the given `initialToken` in the auth header.
+ * This fetch function automatically detects expired tokens and refreshes them using the given `tokenRefreshHandler`.
+ * It buffers all requests until the token is refreshed.
+ *
+ * @param initialToken - The initial token to use.
+ * @param tokenRefreshHandler - The function to call to refresh the token.
+ * @returns A fetch function that refreshes the token when it expires.
+ */
+const createTokenRefreshingFetch = (initialToken: string, tokenRefreshHandler: () => Promise<{ token: string }>) => {
+  let currentToken = initialToken
+  let isRefreshing = false
```

Comment:
This looks unnecessary, `refreshPromise` being nullable already carries this information. With this extra bit of state you need to be careful for it not to go out of sync.

If you find the explicit `isRefreshing` name good for readability, I'd make it a function:

```ts
  const isRefreshing = () => refreshPromise !== null
```